### PR TITLE
Refactor table func interface with const params

### PR DESCRIPTION
**Pull request by ray6080**
# Description

See title.

### CODE REVIEW

File: src/include/function/table_functions.h [hunk 1]
```diff
@@ -73,20 +73,20 @@ struct TableFunctionInitInput {
...
-    std::function<std::unique_ptr<TableFuncSharedState>(TableFunctionInitInput&)>;
+    std::function<std::unique_ptr<TableFuncSharedState>(const TableFunctionInitInput&)>;
 using table_func_init_local_t = std::function<std::unique_ptr<TableFuncLocalState>(
-    TableFunctionInitInput&, TableFuncSharedState*, storage::MemoryManager*)>;
+    const TableFunctionInitInput&, TableFuncSharedState*, storage::MemoryManager*)>;
 using table_func_can_parallel_t = std::function<bool()>;
 using table_func_progress_t = std::function<double(TableFuncSharedState* sharedState)>;
 using table_func_finalize_t =
-    std::function<void(processor::ExecutionContext*, TableFuncSharedState*)>;
+    std::function<void(const processor::ExecutionContext*, TableFuncSharedState*)>;
```
Comment (andyfengHKU) [inline src/include/function/table_functions.h:85]: Mark input as const is fine. I'm slightly worried about marking ExecutionContext & ClientContext as const because some table function might modify it, e.g. if we keep error message in ExecutionContext then `clearWarning` will try to modify it.

No need to revert these changes, we can always change on demand.

**Review by andyfengHKU — APPROVED** [review]

File: src/include/function/table_functions.h (hunk 1 above)
Comment (ray6080) [reply src/include/function/table_functions.h:85]: hmm maybe, but for now I don't think we do that in `finalize`. Let's see how this evolves later.
